### DISCUSSION TITLE
Improving the data from COROS APEX Pro and VERTIX

### DISCUSSION THREAD

**Discussion by Logiqx** [djong1/gp3s#136] [GPS Technology (devices and formats)]
There are a number of issues with the COROS watches and they are listed in a separate project on GitHub - [link](https://github.com/Logiqx/gp3s-coros/issues)

One rather unrealistic aspect of the APEX Pro and VERTIX data is repeated speeds as shown below:

![image](https://user-images.githubusercontent.com/17449078/212746728-51e1c7dc-46a7-4621-b082-a484c8b3abeb.png)

Only 40.201 knots and 40.298 are reported over a period of 10 seconds and this is clearly not what was really happening. This is a really common issue in APEX Pro and VERTIX data and this is completely unrealistic of what happens during a speedsurfing run.

This data issue appears to be an artefact of the Sony GNSS chip used in the COROS APEX Pro and VERTIX. This same artefact is also present in the Garmin Fenix 6 which uses the same Sony GNSS chip (or at least the same family).

Somebody uploaded the technical documentation for the Sony chip to GitHub and I spotted what I think is the likely cause:

https://github.com/Xinyuan-LilyGO/LilyGO-T-Motion/blob/master/datasheet/CXD5603GF_UsersManual_E_013_TianQin.pdf

The issue seems likely to be related to the GNSS positioning algorithm for the special use case ("swinging the receiver by the hand"). This algorithm is enabled by default (see `@GUSE` command) and should not be activitated for speedsurfing or windsurf activities (or activities such as cycling). This seems like the most likely culprit of speeds being repeated in the APEX Pro and VERTIX data, when we would expect them to fluctuate. It may also explain the smoothing that is evident in APEX Pro and VERTIX data and maybe also why 2s results are often lower than other devices.

Perhaps we can collectively ask for a beta firmware with the "swinging the receiver by the hand" filter disabled during windsurfing and speedsurfing activities? It really shouldn't be hard to implement and can potentially make a big difference to the data being logged. If this does turn out to be the case then it would also be beneficial to cycling activities and perhaps some other sports.

**Comment by jhdebruin** [comment]
Good idea, think we should forward this thread (again) to Coros....

**Reply by Logiqx** [reply]
Is this something you can do @jhdebruin?

I'm never quite sure how they regard the e-mails from myself, rather than the actual GP3S team.

**Reply by jhdebruin** [reply]
For sure, we keep pushing :)

**Comment by Logiqx** [comment]
Just adding a comment to say that COROS are hoping to create a beta firmware to test this out.
